### PR TITLE
Update scalameta to 4.3.4 and fix existing tests

### PR DESCRIPTION
**Pull request by tgodzik**

### CODE REVIEW

File: tests/unit/src/test/scala/tests/RenameLspSuite.scala [hunk 1]
```diff
@@ -570,18 +570,30 @@ class RenameLspSuite extends BaseLspSuite("rename") {
...
-  // issue https://github.com/scalameta/scalameta/issues/1169
-  // possibly issue https://github.com/scalameta/scalameta/issues/1845
   renamed(
     "params",
     """|/a/src/main/scala/a/Main.scala
        |case class Name(<<va@@lue>>: String)
        |
        |object Main {
-       |  val name1 = Name(value = "42")
+       |  val name1 = Name(<<value>> = "42")
```
Comment (olafurpg) [inline tests/unit/src/test/scala/tests/RenameLspSuite.scala]: \o/

Comment (tgodzik) [reply tests/unit/src/test/scala/tests/RenameLspSuite.scala]: small victories :smile:

File: tests/unit/src/test/scala/tests/SyntaxErrorLspSuite.scala [hunk 2]
```diff
@@ -102,6 +102,9 @@ class SyntaxErrorLspSuite extends BaseLspSuite("syntax-error") {
         """|a/src/main/scala/Main.scala:1:8: error: identifier expected but 'object' found.
            |object object A
            |       ^^^^^^
+           |a/src/main/scala/Main.scala:2:1: error: <error> is already defined as object <error>
```
Comment (tgodzik) [inline tests/unit/src/test/scala/tests/SyntaxErrorLspSuite.scala]: @olafurpg This looks like a regression, any idea where it came from?

File: tests/unit/src/test/resources/foldingRange/expect/PatternMatching.scala [hunk 3]
```diff
@@ -1,43 +0,0 @@
-class A>>region>>{
```
Comment (tgodzik) [inline tests/unit/src/test/resources/foldingRange/expect/PatternMatching.scala]: PatternMatching test is ignored, which is why save-expect removed it.

File: tests/unit/src/test/scala/tests/SyntaxErrorLspSuite.scala (hunk 2 above)
Comment (tgodzik) [reply tests/unit/src/test/scala/tests/SyntaxErrorLspSuite.scala]: Reported the issue in https://github.com/scalameta/scalameta/issues/2002 - we can take a look at it later.

**Pull request merged**